pandas/core/indexing.py

```diff
@@ -2098,8 +2099,15 @@ def _setitem_with_indexer_missing(self, indexer, value):
                 # We should not cast, if we have object dtype because we can
                 # set timedeltas into object series
                 curr_dtype = self.obj.dtype
-                curr_dtype = getattr(curr_dtype, "numpy_dtype", curr_dtype)
-                new_dtype = maybe_promote(curr_dtype, value)[0]
+                if lib.is_pyarrow_scalar(value) and hasattr(
+                    curr_dtype, "pyarrow_dtype"
+                ):
+                    # TODO promote arrow scalar and type
```

Comment:
this comment is referring to e.g. if `int16[pyarrow]` dtype and value being a too-big integer, right?  it looks like for regular python ints the existing code path might actually work, but that looks fragile.  im thinking the robust way to handle this would be to explicitly handle ArrowEADtype in maybe_promote